core/nodepool/config/templates/stack-template.json

```diff
@@ -101,7 +101,7 @@
               {{end}}
             ],
             "SubnetId": {{$workerSubnet.Ref}},
-            "UserData": {{ $.UserDataWorker.Parts.instance.Template }}
+            "UserData": {{ $.UserDataWorker.RenderInstance true | checkSizeLessThan 16384 | quote }}
```

Comment:
Suppose we now embed the rendered "instance" template which [contains the S3URI](https://github.com/kubernetes-incubator/kube-aws/pull/1220/files#diff-a14eb9921ff28b29aa4b3394378d27d6R21) which contains the hash of the rendered "s3" template, can we safely omit the fingerprint inclusion done in the new `RenderInstance` func?

Reply:
Ah, sorry again and thanks for pointing it out!
I somehow misunderstood that the problem of the inappropriate fingerprint affected worker nodes only but that's not true, indeed.
That being said, I now believe that inserting a "correct" fingerprint through `RenderInstance` is completely valid.

Regarding the API call minimization thing, I still wonder if what [I have suggested above](https://github.com/kubernetes-incubator/kube-aws/pull/1220/files#r179750922) is possible. If it doesn't work, yeah, it would be a hard journey... In that case, I'm open to drop the API call minimization trick for worker nodes for now and discuss about bring it back in an another issue.